loader/loader_environment.c

```diff
@@ -522,10 +522,12 @@ VkResult loader_add_environment_layers(struct loader_instance *inst, const enum
 
         // If not a meta-layer, simply add it.
         if (0 == (source_prop->type_flags & VK_LAYER_TYPE_FLAG_META_LAYER)) {
-            loader_add_layer_properties_to_list(inst, target_list, 1, source_prop);
-            loader_add_layer_properties_to_list(inst, expanded_target_list, 1, source_prop);
+            res = loader_add_layer_properties_to_list(inst, target_list, 1, source_prop);
+            if (res == VK_ERROR_OUT_OF_HOST_MEMORY) return res;
```

Comment:
Shouldn't you goto out here?  I.e. woun't have have a few things that still need to be freed?